src/dev/impl/DevToys/ViewModels/Tools/Generators/CheckSumGenerator/CheckSumGeneratorToolViewModel.cs

```diff
@@ -0,0 +1,429 @@
+                if (storageItem is StorageFile file)
+                {
+                    InputFile = file;
+                }
+            }).ConfigureAwait(false);
+        }
+
+        #endregion
+
+        #region SelectFilesBrowseCommand
+
+        public IAsyncRelayCommand SelectFilesBrowseCommand { get; }
+
+        private async Task ExecuteSelectFilesBrowseCommandAsync()
+        {
+            var filePicker = new FileOpenPicker
+            {
+                ViewMode = PickerViewMode.List,
+                SuggestedStartLocation = PickerLocationId.ComputerFolder,
+            };
+            filePicker.FileTypeFilter.Add("*");
+
+            await ThreadHelper.RunOnUIThreadAsync(async () =>
+            {
+                StorageFile file = await filePicker.PickSingleFileAsync();
+                if (file != null)
+                {
+                    InputFile = file;
+                }
+            });
+        }
+
+        #endregion
+
+        #region SelectFilesPasteCommand
+
+        public IAsyncRelayCommand SelectFilesPasteCommand { get; }
+
+        private async Task ExecuteSelectFilesPasteCommandAsync()
+        {
+            await ThreadHelper.RunOnUIThreadAsync(async () =>
+            {
+                DataPackageView? dataPackageView = Clipboard.GetContent();
+
+                IStorageItem? storageItem = await ExtractStorageItem(dataPackageView);
+                if (storageItem is StorageFile file)
+                {
+                    InputFile = file;
+                }
+            }).ConfigureAwait(false);
+        }
+
+        #endregion
+
+        #region CancelCommand
+
+        public IRelayCommand CancelCommand { get; }
+
+        private void ExecuteCancelCommand()
+        {
+            lock (_lockObject)
+            {
+                _hasCancelledCalculation = true;
+                _cancellationTokenSource.Cancel();
+                _cancellationTokenSource.Dispose();
+
+                _cancellationTokenSource = new CancellationTokenSource();
+
+                if (_outputHashingAlgorithm != null)
+                {
+                    InputHashingAlgorithm = _outputHashingAlgorithm;
+                }
+
+                _isCalculationInProgress = false;
+                _hasCancelledCalculation = false;
+                ShouldDisplayProgress = false;
+                ResetProgress();
+            }
+        }
+
+        #endregion
+
+        private void UpdateOutputCase()
+        {
+            if (string.IsNullOrWhiteSpace(Output))
+            {
+                return;
+            }
+
+            if (IsUppercase)
+            {
+                Output = Output!.ToUpperInvariant();
+            }
+            else
+            {
+                Output = Output!.ToLowerInvariant();
+            }
+        }
+
+        private void CheckSum()
+        {
+            lock (_lockObject)
+            {
+                ResetProgress();
```

Comment:
I played a little bit with the tool and found an issue in the following scenario:
1. Select a large file.
2. The hash calculation is in progress, the progress ring is increasing. While this time, select another file.

When doing this, the previous calculation process isn't canceled. You might be able to fix it by calling invoking the `CancelCommand` instread of `ResetProgress`.

Reply:
Good catch. Initially was invoking the `CancelCommand` on each calculation, but it was messing up the logic to revert back the dropdown to the previous hashing algorithm if you cancel the calculation.
I'm now passing an argument that specifies if we should revert to the previous hash algorithm or not.
It should be fixed now.